### PR TITLE
Add rosetta module to maven cycle

### LINKED ISSUE #1523: Integrate rosetta module into maven cycle
**Problem**
The rosetta module current uses standalone go commands to build and test.
Given the rest of the modules can all be managed under maven for build and test we should incorporate rosetta also.

**Solution**
Integrate rosetta module into maven cycle [mvn-golang](https://github.com/raydac/mvn-golang) seems like a good starting option

**Alternatives**

**Additional Context**

### PR DESCRIPTION
**Pull request by Nana-EC**
**Detailed description**:
Rosetta module isn't currently included in the mane build cycle

- Add `mvn-golang-wrapper` plugin to add maven cycle support to rosetta module
- Update`rosetta-api.yml` GitHub action to run maven package

**Which issue(s) this PR fixes**:
Fixes #1523 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: hedera-mirror-rosetta/pom.xml [hunk 1]
```diff
@@ -20,11 +20,45 @@
...
+                    <execution>
+                        <id>default-test</id>
+                        <configuration>
+                            <buildFlags>
+                                <flag>-v</flag>
+                            </buildFlags>
+                            <packages>
+                                <package>${project.basedir}/...</package>
+                            </packages>
+                            <outLogFile>${project.artifactId}-test.log</outLogFile>
```
Comment (steven-sheehy) [inline hedera-mirror-rosetta/pom.xml:47]: We should log to files. We don't for any other build.

File: .github/workflows/rosetta-api.yml [hunk 2]
```diff
@@ -24,14 +24,7 @@ jobs:
...
       - name: Build hedera-mirror-rosetta
-        run: go build -i -o rosetta cmd/*
-        working-directory: ${{env.working-directory}}
-      - name: Upload Binary as artifact
-        uses: actions/upload-artifact@master
-        with:
-          name: hedera-mirror-rosetta
-          path: ${{ env.working-directory }}/rosetta
-          if-no-files-found: error
+        run: ./mvnw clean package -pl hedera-mirror-rosetta
```
Comment (steven-sheehy) [inline .github/workflows/rosetta-api.yml:27]: Surprised this works as I thought Maven required Java to be installed.

File: hedera-mirror-rosetta/pom.xml [hunk 3]
```diff
@@ -6,7 +6,7 @@
     <description>Rosetta REST API for the Hedera Mirror Node</description>
```
Comment (steven-sheehy) [inline hedera-mirror-rosetta/pom.xml:6]: You should disable rosetta in CircleCI.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rosetta/pom.xml (hunk 3 above)
Comment (Nana-EC) [reply hedera-mirror-rosetta/pom.xml:6]: Skipped for circle ci build

File: hedera-mirror-rosetta/pom.xml (hunk 1 above)
Comment (Nana-EC) [reply hedera-mirror-rosetta/pom.xml]: Removed

**Review by steven-sheehy — APPROVED** [review]

File: hedera-mirror-rosetta/pom.xml [hunk 4]
```diff
@@ -20,11 +20,44 @@
     </properties>
 
     <build>
+        <sourceDirectory>${project.basedir}/cmd</sourceDirectory>
```
Comment (xin-hedera) [inline hedera-mirror-rosetta/pom.xml:23]:
```suggestion
        <sourceDirectory>${project.basedir}{file.separator}cmd</sourceDirectory>
```

File: hedera-mirror-rosetta/pom.xml [hunk 5]
```diff
@@ -20,11 +20,44 @@
...
+                <extensions>true</extensions>
+                <executions>
+                    <execution>
+                        <id>default-test</id>
+                        <configuration>
+                            <buildFlags>
+                                <flag>-v</flag>
+                            </buildFlags>
+                            <packages>
+                                <package>${project.basedir}/...</package>
```
Comment (xin-hedera) [inline hedera-mirror-rosetta/pom.xml:45]:
```suggestion
                                <package>${project.basedir}{file.separator}...</package>
```

File: hedera-mirror-rosetta/pom.xml [hunk 6]
```diff
@@ -20,11 +20,44 @@
...
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>default-build</id>
+                        <configuration>
+                            <buildFlags>
+                                <flag>-a</flag>
+                                <flag>-i</flag>
+                            </buildFlags>
+                            <sources>${project.basedir}/cmd</sources>
```
Comment (xin-hedera) [inline hedera-mirror-rosetta/pom.xml:56]:
```suggestion
                            <sources>${project.basedir}${file.separator}cmd</sources>
```

Comment (Nana-EC) [reply hedera-mirror-rosetta/pom.xml:56]: Good suggestion, updated

File: hedera-mirror-rosetta/pom.xml (hunk 5 above)
Comment (Nana-EC) [reply hedera-mirror-rosetta/pom.xml]: Good suggestion, updated

File: hedera-mirror-rosetta/pom.xml (hunk 4 above)
Comment (Nana-EC) [reply hedera-mirror-rosetta/pom.xml]: Good suggestion, updated

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**